Shared/Realm.Shared/Realm.cs

```diff
@@ -301,7 +301,19 @@ public override int GetHashCode()
         /// <param name="configuration">A configuration which supplies the realm path.</param>
         public static void DeleteRealm(RealmConfiguration configuration)
         {
-            configuration.DeleteRealm();
+            // TODO add cache checking when implemented, https://github.com/realm/realm-dotnet/issues/308
+            // when cache checking, uncomment in IntegrationTests.cs RealmInstanceTests.DeleteRealmFailsIfOpenSameThread and add a variant to test open on different thread
+            var lockOnWhileDeleting = new object();
+            lock (lockOnWhileDeleting)
```

Comment:
I know this is just moving things to where they were, but this lock seems wrong, do you feel like refactoring it or adding a TODO to fix it later? :)